bpf/kmesh/probes/performance_probe.h

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+/* Copyright Authors of Kmesh */
+
+#ifndef __KMESH_BPF_PERFORMANCE_MONITOR_H__
+#define __KMESH_BPF_PERFORMANCE_MONITOR_H__
+
+#include "bpf_common.h"
+
+enum {
+    SOCK_TRAFFIC_CONTROL = 1,
+    XDP_SHUTDOWN = 2,
+    ENABLE_ENCODING_METADATA = 3,
+};
+
+struct operation_usage_data {
+    __u64 start_time;
+    __u64 end_time;
+    __u32 operation_type;
+};
+
+struct operation_usage_key {
+    __u32 tid;
+    __u32 operation_type;
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, struct operation_usage_key);
+    __type(value, struct operation_usage_data);
+    __uint(max_entries, 1024);
+} performance_data_map SEC(".maps");
+
+struct {
+    __uint(type, BPF_MAP_TYPE_RINGBUF);
+    __uint(max_entries, RINGBUF_SIZE);
+} map_perf_info SEC(".maps");
```

Comment:
why not remove `performance_data_map` map and make it inlined into the ringbuffer?